crates/nu-parser/src/parser.rs

```diff
@@ -1250,68 +1250,66 @@ pub fn parse_internal_call(
     }
 }
 
-pub fn parse_call(working_set: &mut StateWorkingSet, spans: &[Span], head: Span) -> Expression {
-    trace!("parsing: call");
-
-    if spans.is_empty() {
-        working_set.error(ParseError::UnknownState(
-            "Encountered command with zero spans".into(),
-            Span::concat(spans),
-        ));
-        return garbage(working_set, head);
-    }
-
+/// finds the longest command matching a command name and a list of arguments
+/// if a command is found, it returns its decl id and the number of arguments that were consumed as
+/// part of the command name
+pub fn find_longest_command(
```

Comment:
Nice cleanup, I like this

Reply:
I'm glad you like it, I was worried it is making things more convoluted :sweat_smile: